src/views/News/components/NewsList/index.js

```diff
@@ -1,164 +1,120 @@
-    const { personalUnapprovedNews } = this.props;
-    let postings;
-    let personalUnapprovedPostings;
-    let header;
-
-    const headerStyle = {
-      backgroundColor: gordonColors.primary.blue,
-      color: '#FFF',
-      padding: '10px',
-    };
-
-    /********** HEADER ***********/
-    // Show single 'news' column for narrrow viewports
-    if (window.innerWidth < this.breakpointWidth) {
-      personalUnapprovedPostings = personalUnapprovedNews.map((posting) => (
-        <NewsItem
-          posting={posting}
-          key={posting.SNID}
-          size="single"
-          updateSnackbar={this.props.updateSnackbar}
-          currentUsername={this.props.currentUsername}
-          callFunction={this.props.callFunction}
-          unapproved
-        />
-      ));
-
-      postings = news.map((posting) => (
-        <NewsItem
-          posting={posting}
-          key={posting.SNID}
-          size="single"
-          updateSnackbar={this.props.updateSnackbar}
-          currentUsername={this.props.currentUsername}
-          callFunction={this.props.callFunction}
-        />
-      ));
-
-      header = (
-        <div style={headerStyle}>
-          <Grid container direction="row">
-            <Grid item xs={12}>
-              <Typography variant="body2" style={headerStyle}>
-                NEWS
-              </Typography>
-            </Grid>
-          </Grid>
-        </div>
-      );
-    }
-
-    // Show full news columns in header for larger viewports
-    else if (news) {
-      personalUnapprovedPostings = personalUnapprovedNews.map((posting) => (
-        <NewsItem
-          posting={posting}
-          key={posting.SNID}
-          size="full"
-          updateSnackbar={this.props.updateSnackbar}
-          currentUsername={this.props.currentUsername}
-          callFunction={this.props.callFunction}
-          unapproved
-        />
-      ));
-
-      postings = news.map((posting) => (
-        <NewsItem
-          posting={posting}
-          key={posting.SNID}
-          updateSnackbar={this.props.updateSnackbar}
-          currentUsername={this.props.currentUsername}
-          callFunction={this.props.callFunction}
-          size="full"
-        />
-      ));
-
-      header = (
-        <Grid container direction="row" style={headerStyle}>
-          <Grid item xs={2}>
-            <Typography variant="body1" style={headerStyle}>
-              CATEGORY
-            </Typography>
-          </Grid>
-          <Grid item xs={5}>
-            <Typography variant="body1" style={headerStyle}>
-              SUBJECT
-            </Typography>
-          </Grid>
-          <Grid item xs={3}>
-            <Typography variant="body1" style={headerStyle}>
-              POSTED BY
-            </Typography>
-          </Grid>
-          <Grid item xs={2}>
-            <Typography variant="body1" style={headerStyle}>
-              POSTED
+  if (window.innerWidth < BREAKPOINT_WIDTH) {
```

Comment:
Also, the way this is currently, this will not automatically resize if the window changes size. It will only check every time it renders, and since you don't have anything watching for a resize event to cause it to rerender, it may not behave as expected.

Reply:
> 
> 
> The control flow here makes very little sense. You have if it's a small screen, show the small version, else if there's news to show, show the desktop version.
> 
> I think it should be
> 
> ```js
> if (news || personalUnapprovedNews) {
>     if (window.innerWidth < BREAKPOINT_WIDTH) {
>         // Show the mobile version
>     } else {
>         // Show the desktop version
>     }
> } else {
>     // Show some message about "no news to show"
> }
> ```

I agree- it makes no sense. When I came to the project it was designed this way, but although it behaves correctly, it doesn't read correctly. I will change it.

Reply:
> 
> 
> Also, the way this is currently, this will not automatically resize if the window changes size. It will only check every time it renders, and since you don't have anything watching for a resize event to cause it to rerender, it may not behave as expected.

Correct. The old class component used forceUpdate before, but when I refactored it, I wasn't sure how to convert this functionality, so I dropped that.

Reply:
This will be fixed in the next commit!